test/NuGet.Core.Tests/NuGet.Common.Test/PathResolverTests.cs

```diff
@@ -0,0 +1,703 @@
+        }
+
+        [Platform(Platform.Windows)]
+        [Theory]
+        [InlineData("dir1/**/file2.txt")]
+        [InlineData(@"dir1\**\file2.txt")]
+        public void PathResolver_PerformWildcardSearch_WithDirectoryGlobstarAndFileNameAtNonRootDirectoryRecursivelyFindsAllMatchingFiles_OnWindows(string searchPath)
+        {
+            var actualFullPaths = PathResolver.PerformWildcardSearch(_fixture.TestDirectoryPath, searchPath);
+
+            Verify(new[]
+                {
+                    @"\dir1\file2.txt",
+                    @"\dir1\dir2\file2.txt",
+                    @"\dir1\dir2\dir3\file2.txt",
+                    @"\dir1\dir4\file2.txt"
+                }, actualFullPaths);
+        }
+
+        [Platform(Platform.Darwin)]
+        [Theory]
+        [InlineData("dir1/**/file2.txt", new[]
+            {
+                "/dir1/file2.txt",
+                "/dir1/dir2/file2.txt",
+                "/dir1/dir2/dir3/file2.txt",
+                "/dir1/dir4/file2.txt"
+            })]
+        [InlineData(@"dir1\**\file2.txt", new string[] { })]
+        public void PathResolver_PerformWildcardSearch_WithDirectoryGlobstarAndFileNameAtNonRootDirectoryRecursivelyFindsAllMatchingFiles_OnMacOs(string searchPath, string[] expectedResults)
+        {
+            var actualFullPaths = PathResolver.PerformWildcardSearch(_fixture.TestDirectoryPath, searchPath);
+
+            Verify(expectedResults, actualFullPaths);
+        }
+
+        [Platform(Platform.Linux)]
+        [Theory]
+        [InlineData("dir1/**/file2.txt", new[]
+            {
+                "/dir1/file2.txt",
+                "/dir1/dir2/file2.txt",
+                "/dir1/dir2/dir3/file2.txt",
+                "/dir1/dir4/file2.txt"
+            })]
+        [InlineData(@"dir1\**\file2.txt", new string[] { })]
+        public void PathResolver_PerformWildcardSearch_WithDirectoryGlobstarAndFileNameAtNonRootDirectoryRecursivelyFindsAllMatchingFiles_OnLinux(string searchPath, string[] expectedResults)
+        {
+            var actualFullPaths = PathResolver.PerformWildcardSearch(_fixture.TestDirectoryPath, searchPath);
+
+            Verify(expectedResults, actualFullPaths);
+        }
+
+        [Platform(Platform.Windows)]
+        [Theory]
+        [InlineData("dir1/dir*/*.txt")]
+        [InlineData(@"dir1\dir*\*.txt")]
+        public void PathResolver_PerformWildcardSearch_WithDirectoryPatternAndFileNamePatternRecursivelyFindsAllMatchingFiles_OnWindows(string searchPath)
+        {
+            var actualFullPaths = PathResolver.PerformWildcardSearch(_fixture.TestDirectoryPath, searchPath);
+
+            Verify(new[]
+                {
+                    @"\dir1\dir2\file1.txt",
+                    @"\dir1\dir2\file2.txt",
+                    @"\dir1\dir4\file1.txt",
+                    @"\dir1\dir4\file2.txt"
+                }, actualFullPaths);
+        }
+
+        [Platform(Platform.Darwin)]
```

Comment:
This looks the same as the Linux test, can a `[Theory]` have two `[Platform]` attributes to act as an `or`?